staging/src/k8s.io/client-go/plugin/pkg/client/auth/gcp/gcp.go

```diff
@@ -153,6 +166,19 @@ func tokenSource(isCmd bool, gcpConfig map[string]string) (oauth2.TokenSource, e
 
 	// Google Application Credentials-based token source
 	scopes := parseScopes(gcpConfig)
+	// If "credentials-file" is set, load from file. Otherwise, use default
+	// token source.
+	if credentialsFile, ok := gcpConfig["credentials-file"]; ok {
+		data, err := ioutil.ReadFile(credentialsFile)
```

Comment:
This is evaluated relative to the CWD of the kubectl which is strange. Should we disallow relative paths?